plugins/phpmyadmin-rule-exclusions-before.conf

```diff
@@ -336,159 +347,541 @@ SecAction \
     pass,\
     t:none,\
     nolog,\
     ver:'phpmyadmin-rule-exclusions-plugin/1.0.0',\
-    chain"
-    SecRule &REQUEST_COOKIES_NAMES:'/^(?:phpMyAdmin|phpMyAdmin_https)$/' "@eq 1" \
-        "t:none,\
-        ctl:ruleRemoveTargetById=942140;ARGS:db"
+    ctl:ruleRemoveTargetById=942190;ARGS,\
+    ctl:ruleRemoveTargetById=942470;ARGS"
+
+# Structure -> Change columns
+# Column types contain SQL keyword. One commonly adds multiple columns.
+# The column types are in ARGS:field_type[n] and ARGS:field_type_orig[n].
+SecRule ARGS:route "@streq /table/structure/change" \
+    "id:9513790,\
+    phase:1,\
+    pass,\
+    t:none,\
+    nolog,\
+    ver:'phpmyadmin-rule-exclusions-plugin/1.0.0',\
+    ctl:ruleRemoveTargetById=942190;ARGS,\
+    ctl:ruleRemoveTargetById=942470;ARGS,\
+    ctl:ruleRemoveTargetById=942140;ARGS:db,\
+    ctl:ruleRemoveTargetById=942100;ARGS:field_extra_orig[1]"
+
+# Query -> Multi-table query
+SecRule ARGS:route "@streq /database/multi-table-query" \
+    "id:9513800,\
+    phase:1,\
+    pass,\
+    t:none,\
+    nolog,\
+    ver:'phpmyadmin-rule-exclusions-plugin/1.0.0',\
+    ctl:ruleRemoveTargetByTag=attack-sqli;ARGS:sql_query,\
+    ctl:ruleRemoveTargetById=932115;ARGS:sql_query"
+
+# Query -> Query by example
+SecRule ARGS:route "@streq /database/qbe" \
+    "id:9513810,\
+    phase:1,\
+    pass,\
+    t:none,\
+    nolog,\
+    ver:'phpmyadmin-rule-exclusions-plugin/1.0.0',\
+    ctl:ruleRemoveTargetByTag=attack-sqli;ARGS:sql_query,\
+    ctl:ruleRemoveTargetById=932115;ARGS:sql_query"
+
+# Setup
+SecRule ARGS:route "@streq /config/set" \
+    "id:9513820,\
+    phase:1,\
+    pass,\
+    t:none,\
+    nolog,\
+    ver:'phpmyadmin-rule-exclusions-plugin/1.0.0',\
+    ctl:ruleRemoveTargetById=942110;ARGS:value"
+
+# Databases, sorting
+SecRule ARGS:route "@streq /server/databases" \
+    "id:9513830,\
+    phase:1,\
+    pass,\
+    t:none,\
+    nolog,\
+    ver:'phpmyadmin-rule-exclusions-plugin/1.0.0',\
+    ctl:ruleRemoveTargetById=942140;ARGS:sort_by"
+
+# Session cookies whitelist + persistent cookies
+# These cookies may persist beyond sessions and may block a user when trying
+# to access phpMyAdmin for a second time.
+SecAction \
+    "id:9513840,\
+    phase:1,\
+    pass,\
+    t:none,\
+    nolog,\
+    ver:'phpmyadmin-rule-exclusions-plugin/1.0.0',\
+    ctl:ruleRemoveTargetById=941100;REQUEST_COOKIES,\
+    ctl:ruleRemoveTargetById=941120;REQUEST_COOKIES,\
+    ctl:ruleRemoveTargetById=942100;REQUEST_COOKIES:auto_saved_sql,\
+    ctl:ruleRemoveTargetById=942200;REQUEST_COOKIES:pmaAuth-1,\
+    ctl:ruleRemoveTargetById=942340;REQUEST_COOKIES:pmaAuth-1,\
+    ctl:ruleRemoveTargetById=942200;REQUEST_COOKIES:pmaUser-1,\
+    ctl:ruleRemoveTargetById=942340;REQUEST_COOKIES:pmaUser-1,\
+    ctl:ruleRemoveTargetById=942200;REQUEST_COOKIES:pma_console_config,\
+    ctl:ruleRemoveTargetById=942260;REQUEST_COOKIES:pma_console_config"
+
+SecMarker "END-PHPMYADMIN-RULE-EXCLUSIONS-PLUGIN-V51-URL-FORMAT"
```

Comment:
Can you limit the URLs that can be called to only the endpoints that 5.1 has?
https://github.com/phpmyadmin/phpmyadmin/blob/QA_5_1/index.php
https://github.com/phpmyadmin/phpmyadmin/blob/QA_5_1/url.php
https://github.com/phpmyadmin/phpmyadmin/blob/QA_5_1/show_config_errors.php
https://github.com/phpmyadmin/phpmyadmin/blob/QA_5_1/js/messages.php
PHP files in https://github.com/phpmyadmin/phpmyadmin/tree/QA_5_1/setup

See my nginx regex https://github.com/sudo-bot/gh-deployer-container/blob/04b50cde53ac13f71c444b09dd31dcecaca3eee2/docker/nginx-default.conf#L29 

```nginx
location ~ ^(?!(\/favicon\.ico|\/robots\.txt|\/index\.php|\/url\.php|\/show_config_errors\.php|\/js\/messages\.php|\/js\/dist|\/js\/vendor|\/doc\/html|\/setup|\/themes|\/)) {
        deny all;
    }
```

Reply:
@williamdes Not sure about this. What exactly do you expect from this?

Reply:
it limits the urls that can be visited adding more security by 403 everything else

Reply:
Creating an allow-list is a useful security measure, but it's out of scope for a rule exclusion plugin.

If you guys think there is wider interest for such a configuration, I suggest to create a separate allow-list plugin.